build/main.go

```diff
@@ -43,7 +44,25 @@ func main() {
 			build.Failure(103)
 			return
 		}
+		build.Success()
+		return
+	}
+
+	if invoker, ok, err := command.NewCommandInvoker(build); err != nil {
+		build.Logger.Info(err.Error())
+		build.Failure(102)
+		return
+	} else if ok {
+		if err = invoker.Contribute(); err != nil {
+			build.Logger.Info(err.Error())
+			build.Failure(103)
+			return
+		}
+		build.Success()
+		return
 	}
 
-	build.Success()
+	build.Logger.Info("Buildpack passed detection but did not know how to actually build. Should never happen.")
```

Comment:
Not sure I follow.

An app that has just a toml file but would fail to qualify as a java or command app would NOT pass the `detect` phase (and hence never get there in the `build` phase)